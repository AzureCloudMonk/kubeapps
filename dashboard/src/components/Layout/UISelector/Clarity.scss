.clr-row {
  // By default, a row has negative margin. This causes
  // that a row for the whole body surpasses its limits
  margin: 0;
}
