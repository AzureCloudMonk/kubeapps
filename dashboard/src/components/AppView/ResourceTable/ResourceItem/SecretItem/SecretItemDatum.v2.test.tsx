import { mount, shallow } from "enzyme";
import * as React from "react";

import { CdsIcon } from "components/Clarity/clarity";
import SecretItemDatum from "./SecretItemDatum.v2";

const testProps = {
  name: "foo",
  value: "YmFy", // foo
};

it("renders the secret datum (hidden by default)", () => {
  const wrapper = mount(<SecretItemDatum {...testProps} />);
  expect(wrapper.find(CdsIcon).prop("shape")).toBe("eye");
  expect(wrapper).toMatchSnapshot();
});

it("displays the secret datum value when clicking on the icon", () => {
  const wrapper = mount(<SecretItemDatum {...testProps} />);
  expect(wrapper.find(".secret-datum-text").text()).toContain("foo: 3 bytes");
  const icon = wrapper.find("button");
  icon.simulate("click");
  wrapper.update();
  expect(wrapper.find(CdsIcon).prop("shape")).toBe("eye-hide");
  expect(wrapper.find(".secret-datum-text").text()).toContain("foo: bar");
});
