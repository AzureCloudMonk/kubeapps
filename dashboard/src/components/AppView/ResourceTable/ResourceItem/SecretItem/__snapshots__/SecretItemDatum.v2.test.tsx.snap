// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the secret datum (hidden by default) 1`] = `
<SecretItemDatum
  name="foo"
  value="YmFy"
>
  <Row
    list={false}
  >
    <div
      className="clr-row"
    >
      <Column
        listItem={false}
        span={1}
      >
        <div
          className="clr-col-1"
        >
          <button
            aria-controls="secret-item-datum-foo-ref"
            aria-expanded={false}
            aria-label="Show Secret"
            className="secret-datum-icon"
            onClick={[Function]}
          >
            <ReactWrapperComponent
              shape="eye"
              size="md"
              solid={true}
            >
              <cds-icon />
            </ReactWrapperComponent>
          </button>
        </div>
      </Column>
      <Column
        listItem={false}
        span={11}
      >
        <div
          className="clr-col-11"
        >
          <div
            className="secret-datum-text"
            id="secret-item-datum-foo-ref"
          >
            <span>
              foo
              : 
              3 bytes
            </span>
          </div>
        </div>
      </Column>
    </div>
  </Row>
</SecretItemDatum>
`;
